internal/testutil/datagen/datagen.sh

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env bash
+
+set -e
+
+# scrapper_selector is the label with which the scraper deployment is deployed.
+scrapper_selector="app=scraper"
+
+# main subcommand runs the whole flow of the script: Bootstrap, scrape, and cleanup
+function main() {
+    if [[ -z "$1" ]]; then
+        echo "Usage: $0 <output_folder>"
+        exit 1
+    fi
+
+    bootstrap
+
+    pod=$(scraper_pod "$scrapper_selector")
+    if [[ -z "pod" ]]; then
+        echo "Could not find scraper pod (-l $scrapper_selector)"
+        exit 1
+    fi
+
+    mkdir -p "$1" && cd "$1"
+    testinfo > README.md
+
+    # Kubelet endpoints
+    mkdir -p kubelet
+    echo "Extracting kubelet /pods"
+    kubectl exec -n scraper $pod -- datagen.sh scrape kubelet pods > kubelet/pods
```

Comment:
we could check if `minikube tunnel` would cover all use cases. I though it did not, but running it against the api-server seems reaching it:
```
curl -X GET https://192.168.183.33:8443/metrics --header "Authorization: Bearer $TOKEN" --insecure
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {

  },
  "status": "Failure",
  "message": "forbidden: User \"system:anonymous\" cannot get path \"/metrics\"",
  "reason": "Forbidden",
  "details": {

  },
  "code": 403
}%


However, no idea if you prefer to run it inside the cluster or having to discover the IP, moreover in that case we would be tied to minikube


```

Reply:
I tried to choose a balance between minikube defaults (because it's what we use) and something that could be useful in other environments if they are not too harsh. The scraper pod is ugly but I think it will work in more scenarios, like kubeadm and even most managed envs (ofc without the control plane metrics).